File: programmers/42587.cpp
// 3:32-4:10
#include <string>
#include <vector>
#include <deque>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 1;
    deque<int> q;
    bool flag;
    int temp;

    for (int i : priorities) q.push_back(i);

    while (!q.empty()) {
        flag = false;
        for (int j : q) {
            if (j > q.front()) flag = true; // max_element로 대체가능했음!!
        }
        if (flag == true) {
            temp = q.front();
            q.pop_front();
            q.push_back(temp);
            if (location == 0) location = q.size();
            location--;
        }
        else {
            if (location == 0) return answer;
            q.pop_front();
            location--;
            answer++;
        }

    }

}

File: programmers/42587.py
from collections import deque

def solution(priorities, location):
    answer = 1
    de = deque(priorities)
    max_val = max(de)

    while True:
        if de[0] < max_val:
            de.append(de.popleft())
            if location == 0:
                location = len(de)
            location -= 1
        else:
            if location == 0:
                return answer
            de.popleft()
            location -= 1
            answer += 1
            max_val = max(de)